src/main/java/com/iexec/worker/docker/CustomDockerClient.java

```diff
@@ -199,91 +221,78 @@ private HostConfig createHostConfig(Map<String, String> bindPaths, boolean isSgx
                 .build();
     }
 
-    /**
-     * This manages the docker execution workflow.
-     * We create and start the container.
-     * 
-     * if (maxExecutionTime == 0)
-     *      the container will be considered a service
-     *      and will run without a timeout (such as the LAS service)
-     * 
-     * if (maxExecutionTime != 0)
-     *      we wait for the execution to end or we stop the container
-     *      when it reaches the deadline.
-     * 
-     * In the latter case we return the stdout logs as a string.
-     */
-    private String runContainerAndGetLogs(String containerName, ContainerConfig containerConfig, long maxExecutionTime) {
-        String containerId = runContainer(containerName, containerConfig);
-        Date executionTimeoutDate = Date.from(Instant.now().plusMillis(maxExecutionTime));
-        waitContainer(containerName, containerId, executionTimeoutDate);
-        log.info("End of execution [containerName:{}]", containerName);
-
-        // docker container stop
-        stopContainer(containerId);
-
-        // docker container logs
-        String stdout = getContainerLogs(containerId);
-
-        // docker container rm
-        removeContainer(containerId);
-        return stdout;
-    }
-
-    public String runContainer(String containerName, ContainerConfig containerConfig) {
+    public CustomContainerInfo createAndStartContainer(String containerName, ContainerConfig containerConfig) {
         if (containerConfig == null) {
-            log.error("Could not run container, container config is null [containerName:{}]", containerName);
-            return "";
+            log.error("Cannot create container since config is null [containerName:{}]", containerName);
+            return null;
```

Comment:
We should avoid to return null, can we return an optional instead?

Reply:
right yeah, thanks!